src/psij/executors/batch/slurm/slurm.mustache

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+{{#job.name}}
+#SBATCH --job-name="{{.}}"
+{{/job.name}}
+
+{{#job.spec.directory}}
+#SBATCH --chdir="{{.}}"
+{{/job.spec.directory}}
+
+{{#job.spec.inherit_environment}}
+#SBATCH --export=ALL
+{{/job.spec.inherit_environment}}
+{{^job.spec.inherit_environment}}
+#SBATCH --export=NONE
+{{/job.spec.inherit_environment}}
+
+{{#job.spec.resources}}
+    {{#exclusive_node_use}}
+#SBATCH --exclusive
+    {{/exclusive_node_use}}
+
+    {{#node_count}}
+#SBATCH --nodes={{.}}
+    {{/node_count}}
+
+    {{#process_count}}
+#SBATCH --ntasks={{.}}
+    {{/process_count}}
+
+    {{#processes_per_node}}
+#SBATCH --ntasks-per-node={{.}}
+    {{/processes_per_node}}
+
+    {{#gpu_cores_per_process}}
+#SBATCH --gpus_per_task={{.}}
+    {{/gpu_cores_per_process}}
+
+    {{#cpu_cores_per_process}}
+        {{!SLURM has different meanings for 'CPU', depending on the exact allocation plugin being
+        used. The default plugin uses the literal meaning for 'CPU' - physical processor. The
+        consumable resource allocation plugin, on the other hand, equates 'CPU' to a thread (if
+        hyperthreaded CPUs are used) or CPU core (for non-hyperthreaded CPUs).}}
+#SBATCH --cpus_per_task={{.}}
+    {{/cpu_cores_per_process}}
+{{/job.spec.resources}}
+
+{{#job.spec.attributes}}
+    {{#duration}}
+#SLURM --time={{#psij.lib.walltime_to_minutes}}{{duration}}{{/psij.lib.walltime_to_minutes}}
+    {{/duration}}
+
+    {{#queue_name}}
+#SLURM --partition="{{.}}"
+    {{/queue_name}}
+
+    {{#project_name}}
+#SLURM --account="{{.}}"
+    {{/project_name}}
+
+    {{#reservation_id}}
+#SLURM --reservation="{{.}}"
+    {{/reservation_id}}
+
+    {{#custom_attributes.slurm}}
+#SBATCH --{{key}}="{{value}}"
+    {{/custom_attributes.slurm}}
+
+{{/job.spec.attributes}}
+
+{{!redirect output here instead of through #SBATCH directive since SLURM_JOB_ID is not available
```

Comment:
If you want, I think it is possible to use the `%j` filename pattern supported by Slurm here: https://slurm.schedmd.com/sbatch.html#SECTION_%3CB%3Efilename-pattern%3C/B%3E

Reply:
Thanks. Good point. However, that might be harder to port to other schedulers and ultimately doesn't buy much: errors with directives should cause sbatch to fail to submit, so there is no useful output from the submit script before the redirect command.